src/Features/CSharp/Portable/Structure/Providers/EventDeclarationStructureProvider.cs

```diff
@@ -30,9 +30,13 @@ internal class EventDeclarationStructureProvider : AbstractSyntaxNodeStructurePr
                 return;
             }
 
+            SyntaxNodeOrToken current = eventDeclaration;
+            var nextSibling = current.GetNextSibling();
+
             spans.AddIfNotNull(CSharpStructureHelpers.CreateBlockSpan(
                 eventDeclaration,
                 eventDeclaration.Identifier,
+                compressEmptyLines: !nextSibling.IsNode || nextSibling.IsKind(SyntaxKind.EventDeclaration) || nextSibling.IsKind(SyntaxKind.EventFieldDeclaration),
```

Comment:
appreciate the grouping of both kinds.

Reply:
It actually doesn't work for an event field followed by a full event (because the field doesn't have a collapsible span) but we at least handle the case we can.